### ISSUE TITLE
`torch.tensor(..., device='meta')` doesn't work

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#55191]
It can be made to work cuz ellipses is an actual Python object, just need to extend the Python binding code a little.

**Comment by venkateshtata** [comment]
Can I work on this as my first issue ? Any suggestions on where to start would be great.

**Comment by ezyang** [comment]
sure. edit `tensor_ctor` in `torch/csrc/utils/tensor_new.cpp`. If device is meta then data is OK to be ellipses, short circuit internal_new_from_data and just directly create an empty meta tensor instead.

**Comment by bdhirsh** [comment]
I'm thinking about picking this back up to reserve for a bootcamper.

@ezyang how would the `torch.tensor(..., device='meta')` know what to set for the tensor's size? Were you thinking of adding an optional size argument, that's only really used in the meta case? (I guess you could assert that it matches data.size() in the normal case).

**Comment by ezyang** [comment]
Yeah, the idea was there would be a size argument. This is a good idea anyway as it is used for regular tensors to disambiguate zero element cases

**Comment by ezyang** [comment]
```
>>> torch.empty((2,0,2))
tensor([], size=(2, 0, 2))
```
